python/ray/autoscaler/kubernetes/example-full.yaml

```diff
@@ -0,0 +1,232 @@
+# An unique identifier for the head node and workers of this cluster.
+cluster_name: default
+
+# The minimum number of workers nodes to launch in addition to the head
+# node. This number should be >= 0.
+min_workers: 0
+
+# The maximum number of workers nodes to launch in addition to the head
+# node. This takes precedence over min_workers.
+max_workers: 2
+
+# The initial number of worker nodes to launch in addition to the head
+# node. When the cluster is first brought up (or when it is refreshed with a
+# subsequent `ray up`) this number of nodes will be started.
+initial_workers: 0
+
+# Whether or not to autoscale aggressively. If this is enabled, if at any point
+#   we would start more workers, we start at least enough to bring us to
+#   initial_workers.
+autoscaling_mode: default
+
+# The autoscaler will scale up the cluster to this target fraction of resource
+# usage. For example, if a cluster of 10 nodes is 100% busy and
+# target_utilization is 0.8, it would resize the cluster to 13. This fraction
+# can be decreased to increase the aggressiveness of upscaling.
+# This value must be less than 1.0 for scaling to happen.
+target_utilization_fraction: 0.8
+
+# If a node is idle for this many minutes, it will be removed.
+idle_timeout_minutes: 5
+
+# Kubernetes resources that need to be configured for the autoscaler to be
+# able to manage the Ray cluster. If any of the provided resources don't
+# exist, the autoscaler will attempt to create them. If this fails, you may
+# not have the required permissions and will have to request them to be
+# created by your cluster administrator.
+provider:
+    type: kubernetes
+
+    # Exposing external IP addresses for ray pods isn't currently supported.
+    use_internal_ips: true
+
+    # Namespace to use for all resources created.
+    namespace:
```

Comment:
Good question. I originally tried hiding the Kubernetes details from the user but I think it would be less clear as they might not understand exactly how the fields map to Kubernetes resources, we would have to write more code to support different configuration options, and also users would have to learn another new syntax instead of just using the Kubernetes one. I expect that most users will just use the example config, so this shouldn't be much of a problem.

Reply:
Given there isn't much more to a namespace than it's name, I don't think this is particularly true for namespace specifically. I don't think it is uncommon to see configs with a `namespace: some-name` entry. It also matches the kubectl api, e.g., 
```bash
kubectl apply -f config.yaml --namespace some-name
```
That being said, this is a mostly insignificant detail in the grand scheme of things so if you prefer the way it is now, I don't see a problem leaving it as is.

Reply:
That's a good point, would definitely look cleaner that way. I will change it for the namespace field only.